examples/golang/cosmicconnector/README.md

```diff
@@ -0,0 +1,207 @@
+  -log-level info
+```
+
+#### Flags:
+
+- `-config`: Path to the text protobuf configuration file (e.g., `config.textproto`). Required.
+- `-log-level`: Sets the logging level (options: `disabled`, `warn`, `panic`, `info` (default), `fatal`, `error`, `debug`, `trace`).
+- `-dry-run`: (Optional) Validate the configuration without starting the server. Exits with a non-zero return code if the config is invalid.
+
+**Example: Dry Run Configuration Validation**
+
+```bash
+bazel run //examples/golang/cosmicconnector:cosmic_connector -- \
+  -config path/to/config.textproto -dry-run
+```
+
+For more details, refer to the [main.go](./main.go) source file.
+
+## Sample gRPC Calls
+
+Once the Cosmic Connector is running, you can interact with its gRPC services using `grpcurl`. Below are sample commands for various operations.
+
+### ListServiceOptions
+
+Retrieve service options based on specific filters.
+
+```bash
+grpcurl -plaintext -d '{
+  "service_attributes_filters": {
+    "bidirectional_service_attributes_filter": {
+      "bandwidth_bps_minimum": 1000000,
+      "one_way_latency_maximum": "100ms"
+    }
+  }
+}' localhost:50052 outernet.federation.v1alpha.Federation/ListServiceOptions
+```
+
+*Refer to [grpc_server.go](../../pkg/go/server/grpc_server.go) and [handler.go](./handler/handler.go) for implementation details.*
+
+### ScheduleService
+
+Schedule a new service with specified parameters.
+
+```bash
+grpcurl -plaintext -d '{
+  "requestor_edge_to_requestor_edge": {
+    "x_interconnection_points": [{
+      "uuid": "point1",
+      "coordinates": {
+        "entry": [{
+          "interval": {
+            "start_time": "2024-01-01T00:00:00Z",
+            "end_time": "2024-01-02T00:00:00Z"
+          },
+          "geodetic_wgs84": {
+            "longitude_deg": 0,
+            "latitude_deg": 0,
+            "height_wgs84_m": 0
+          }
+        }]
+      }
+    }],
+    "y_interconnection_points": [{
+      "uuid": "point2",
+      "coordinates": {
+        "entry": [{
+          "interval": {
+            "start_time": "2024-01-01T00:00:00Z",
+            "end_time": "2024-01-02T00:00:00Z"
+          },
+          "geodetic_wgs84": {
+            "longitude_deg": 1,
+            "latitude_deg": 1,
+            "height_wgs84_m": 0
+          }
+        }]
+      }
+    }],
+    "directionality": "DIRECTIONALITY_BIDIRECTIONAL"
+  },
+  "service_attributes_filters": {
+    "bidirectional_service_attributes_filter": {
+      "bandwidth_bps_minimum": 1000000,
+      "one_way_latency_maximum": "1s"
+    }
+  },
+  "priority": 1
+}' localhost:50052 outernet.federation.v1alpha.Federation/ScheduleService
+```
+
+*See [handler.go](./handler/handler.go) for the implementation of `ScheduleService`.*
+
+### MonitorServices
+
+Monitor existing services by their IDs.
+
+```bash
+grpcurl -plaintext -d '{"add_service_ids": ["service-1", "service-2"]}' localhost:50052 outernet.federation.v1alpha.Federation/MonitorServices
```

Comment:
I played around with this and noticed that whatever command you send, it'll finish with:

> ERROR:
>   Code: Unknown
>   Message: EOF

which I wouldn't have expected. The results look fine (no output when service isn't active, output when it is), it just always gets this error. Is this expected?